src/main/java/software/amazon/smithy/lsp/language/IdlPosition.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.lsp.language;
+
+import java.util.Optional;
+import org.eclipse.lsp4j.Position;
+import software.amazon.smithy.lsp.project.SmithyFile;
+import software.amazon.smithy.lsp.syntax.Syntax;
+import software.amazon.smithy.lsp.syntax.SyntaxSearch;
+
+sealed interface IdlPosition {
```

Comment:
I hate this class. I spent some time looking for a way to abstract away the direct usage of `documentIndex` and `statementIndex`, both here and in `SyntaxSearch`, but didn't come up with anything great. Open to suggestions.